### ISSUE TITLE
Rule Change: add `punctuation` option to `no-double-punctuation` rule

### ISSUE THREAD

**Issue by lumirlumir** [eslint-markdown/eslint-markdown#762]
Add a `punctuation` option to `no-double-punctuation`, following the option design used by `no-trailing-heading-punctuation`.

## Motivation

`no-double-punctuation` currently always detects pairs made from the built-in ASCII punctuation set (including `?`). The existing `allow` option can exempt individual two-character combinations, but it cannot narrow or replace the set of punctuation characters the rule examines. Projects that intentionally use a smaller punctuation set must enumerate every permitted combination in `allow`.

A `punctuation` option would make the rule configurable at the same level as `no-trailing-heading-punctuation` while preserving the current behavior by default.

## Proposed option

Add a `punctuation: string[]` property to the rule options.

https://github.com/eslint-markdown/eslint-markdown/blob/2d58f54127ff3a1dda55b74119280bdbe3c91a6a/packages/eslint-markdown/src/rules/no-trailing-heading-punctuation.ts#L90-L94

```js
{
  'md/no-double-punctuation': ['error', {
    punctuation: ['.', '!', '?'],
    allow: ['?!', '!!'],
  }],
}
```

- Default: the current `asciiPunctuationWithQuestionMark` list, so existing configurations remain unchanged.
- Validation: an array of unique one-character strings with at least one item, matching the `punctuation` schema of `no-trailing-heading-punctuation`.
- Detection: build the double-punctuation regular expression from the configured `punctuation` list instead of the fixed constant.
- `allow`: continue to apply after detection, and validate each allowed two-character pattern against the configured punctuation list. This keeps `allow` focused on intentional combinations such as `?!`.

## Expected behavior

With `{ punctuation: ['.', '!'] }`:

- `Hello..` and `Hello!!` are reported.
- `Hello?!` is not reported because `?` is outside the configured set.
- `allow: ['!!']` exempts `Hello!!`.
- `allow: ['??']` should be rejected by option validation because `?` is not configured.

## Implementation and test notes

- Reuse `escapeStringRegexp` when constructing the configured character class, as `no-trailing-heading-punctuation` does.
- Keep the existing autofix behavior for duplicate characters and the existing suggestions for mixed characters.
- Add schema/default-option tests and cases covering custom punctuation, `allow` interaction, and backward-compatible defaults.

**Comment by daeun0726** [comment]
@lumirlumir, I'd like to work on this. Could you assign it to me?

**Comment by lumirlumir** [comment]
Thanks for your interest on this issue! A PR is welcome.
